### PR TITLE
Implement ArrayList

### PR DESCRIPTION
**Pull request by soc**

### CODE REVIEW

File: javalib/src/main/scala/java/util/ArrayList.scala [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+          System.arraycopy(other.values, 0, values, index, otherSize)
+          true
+        } else {
+          false
+        }
+      case _ => super.addAll(index, c)
+    }
+  }
+
+  // Needed to avoid linking error
```
Comment (densh) [inline javalib/src/main/scala/java/util/ArrayList.scala]: I've just filed an issue for this: #375 

Can you please change the message `Workaround for #375` ?

File: javalib/src/main/scala/java/util/ListIterator.scala [hunk 2]
```diff
@@ -7,4 +7,9 @@ trait ListIterator[E] extends Iterator[E] {
   def previousIndex(): Int
   def nextIndex(): Int
   def set(e: E): Unit
+
+  // Needed to avoid linking error
```
Comment (densh) [inline javalib/src/main/scala/java/util/ListIterator.scala]: `Workaround for #375`

File: javalib/src/main/scala/java/util/List.scala [hunk 3]
```diff
@@ -1,8 +1,13 @@
 package java.util
 
 trait List[E] extends Collection[E] {
+  def size(): Int
+  def isEmpty(): Boolean
+  def clear(): Unit
+  def iterator(): Iterator[E]
```
Comment (densh) [inline javalib/src/main/scala/java/util/List.scala]: I assume that those are also `Workaround for #375`.

File: javalib/src/main/scala/java/util/AbstractCollection.scala [hunk 4]
```diff
@@ -78,5 +78,5 @@ abstract class AbstractCollection[E] protected () extends Collection[E] {
   }
 
   override def toString(): String =
-    iterator.mkString("[", ",", "]")
+    iterator.mkString("[", ", ", "]")
```
Comment (densh) [inline javalib/src/main/scala/java/util/AbstractCollection.scala]: Why the change? We should stick to what we do on JVM/JS whenever possible.

Comment (soc) [inline javalib/src/main/scala/java/util/AbstractCollection.scala]: Yes, this change moves the implementation in line with what the JDK does.

**Comment by soc** [comment]
Ok, addressed your comments.

**Comment by soc** [comment]
Please drop my name from the CLA list and notify me when it is done.

**Comment by vossad01** [comment]
@densh I am guessing the answer is the obvious, but I figured it doesn't hurt to ask:

Even though the code of this PR (other various such unmerged work) is published under a BSD-3-Clause license, no one can pick up the work of trying to get things integrated due to CLA limitations?

**Comment by densh** [comment]
@vossad01 I'd suggest taking current Scala.js implementation as a reference and pick it up from there.
